### DISCUSSION TITLE
Query array in input receive string if array is only one element ?

### DISCUSSION THREAD

**Discussion by huyhoang160593** [RobinTail/express-zod-api#1388] [Q&A] [answered]
There is pretty annoying things when I define zod array in the get method input(related to the query), when there is more than 1 element, the schema is working just fine, but if there is only 1 element, the schema have detect them as string and not an array, then my schema will fail and throw error :(. Can you provide me a fix for this @RobinTail ?

**Comment by RobinTail** [comment, marked as answer]
Hello @huyhoang160593 

Please read this thread to find out how to encode arrays properly within query parameters, or for alternative ways of transformations:
https://github.com/RobinTail/express-zod-api/discussions/1266#discussioncomment-7428460

In case it does not solve your problem, please provide:
- the code sample of the input schema of your endpoint
- the sample of your request
